eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageCodecs.scala

```diff
@@ -45,8 +45,51 @@ object LightningMessageCodecs {
   // (for something smarter see https://github.com/yzernik/bitcoin-scodec/blob/master/src/main/scala/io/github/yzernik/bitcoinscodec/structures/UInt64.scala)
   val uint64: Codec[Long] = int64.narrow(l => if (l >= 0) Attempt.Successful(l) else Attempt.failure(Err(s"overflow for value $l")), l => l)
 
+  val uint64L: Codec[Long] = int64L.narrow(l => if (l >= 0) Attempt.Successful(l) else Attempt.failure(Err(s"overflow for value $l")), l => l)
```

Comment:
NB: `uint64` codec is only suitable when we now for sure that decoded values will be < 2^63 because java longs are signed. Is it the case here? If not, then we have to use the purpose-build `UINT64` class.

Reply:
Right, I must admit I copy-pasted that part at the beginning and wanted to come back to it, this is a bit unclear. I'll clean that up.

Reply:
As discussed offline, we should probably retire the `uint64` codec and replace it with proper `UInt64` (even though it's a bit more painful to use than `Long`).
I changed this one to really use the 64 bits.